src/LmdbCache.ts

```diff
@@ -30,113 +34,153 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
     }
 
     this.logger.info(`Using location ${cacheOptions.dbLocation}`);
-    this.db = open<V, string>({
+    this.db = open<ClientValueWrapper<V>, string>({
       path: `${cacheOptions.dbLocation}`,
       noSync: cacheOptions.inMemory
     });
   }
 
+  /**
+   * Batch operations are all executed in one transaction using childTransaction.
+   * For each put there is an old entries removal run.
+   */
   async batch(opStack: BatchDBOp<V>[]) {
-    for (const op of opStack) {
-      if (op.type === 'put') {
-        await this.put(op.key, op.value);
-      } else if (op.type === 'del') {
-        await this.delete(op.key);
+    await this.db.childTransaction(async () => {
```

Comment:
not sure if async code within a transactions is safe
<img width="1066" alt="image" src="https://user-images.githubusercontent.com/84311757/232830398-f646a2a5-afec-4f08-b06f-b37ea91cbcab.png">

Reply:
yes, it seems that using transaction or childTransaction functions we may end up with some unpredicted order...
maybe we should switch to transactionSync